R/fit_model.R

```diff
@@ -0,0 +1,179 @@
+#' Fit an EpiNow2 model
+#'
+#' @param data, in the format returned by [CFAEpiNow2Pipeline::read_data()]
+#' @param parameters As returned from
+#'   [CFAEpiNow2Pipeline::read_disease_parameters()]
+#' @param seed The random seed, used for both initialization by EpiNow2 in R and
+#'   sampling in Stan
+#' @param horizon The number of days, as an integer, to forecast
+#' @param priors A list of lists. The first level should contain the key `rt`
+#'   with elements `mean` and `sd` and the key `gp` with element `alpha_sd`.
+#' @param sampler_opts A list. The Stan sampler options to be passed through
+#'   EpiNow2. It has required keys: `cores`, `chains`, `iter_warmup`,
+#'   `iter_sampling`, `max_treedepth`, and `adapt_delta`.
+#'
+#' @return A fitted model object of class `epinow` or, if model fitting fails,
+#'   an NA is returned with a warning
+#' @export
+fit_model <- function(
+    data,
+    parameters,
+    seed,
```

Comment:
Default `seed = NULL`?

Reply:
Let's handle discussion of defaults here: https://github.com/CDCgov/cfa-epinow2-pipeline/pull/26#discussion_r1755154014